tests/test_commands.py

```diff
@@ -236,9 +238,9 @@ def test_sql_comments(self):
 comment */ 4;
 """
         cmd = CrateShell()
-        cmd._exec_and_print = MagicMock()
+        cmd._exec = Mock()
         cmd.process_iterable(sql.splitlines())
-        self.assertListEqual(cmd._exec_and_print.mock_calls, [
+        self.assertListEqual(cmd._exec.mock_calls, [
```

Comment:
That noise, and subsequent spots, have been needed to move the testing probe to a different spot. Those tests verify that what goes in (user input SQL statement) will yield the correct outcome when being submitted to the database.

Previously, the probe was sitting on `_exec_and_print` by mocking it and verifying the calls on this mock. Now, that probing moved to the more low-level `_exec` method. This changes the execution behavior of the test suite, as the `_exec_and_print` method will now be executed fully [^1], unlocking the possibility to test its effects.

[^1]: That is the reason why the test suite also needs to supply a mocked cursor now, because the ingredients of `_exec_and_print` need it.